pkgs/build-support/dart/build-dart-application/generators.nix

```
@@ -71,7 +72,16 @@ let
           runHook preInstall
 
           packageName="$(yq --raw-output .name pubspec.yaml)"
-          jq --arg name "$packageName" '.packages |= . + [{ name: $name, rootUri: "../", packageUri: "lib/" }]' '${packageConfig}' > "$out"
+          languageConstraint="${pubspecLock.sdks.dart}"
+          if [[ "$languageConstraint" =~ ^[[:space:]]*(\^|>=|>)?[[:space:]]*([[:digit:]]+\.[[:digit:]]+)\.[[:digit:]]+.*$ ]]; then
+            languageVersionJson="\"''${BASH_REMATCH[2]}\""
+          elif [ "$languageConstraint" = 'any' ]; then
+            languageVersionJson='null'
+          else
+            # https://github.com/dart-lang/pub/blob/68dc2f547d0a264955c1fa551fa0a0e158046494/lib/src/language_version.dart#L106C35-L106C35
```

```suggestion
            # https://github.com/dart-lang/pub/blob/68dc2f547d0a264955c1fa551fa0a0e158046494/lib/src/language_version.dart#L106
```

I followed this link and was surprised at the empty highlight. Let's just do the whole line.

Also note that on master it's now line 109.